storage/postgres/testdb/testdb.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package testdb creates new databases for tests.
+package testdb
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/trillian/testonly"
+
+	_ "github.com/lib/pq" // mysql driver
+)
+
+var (
+	trillianSQL = testonly.RelativeToPackage("../storage.sql")
+	dataSource  = "database=test sslmode=disable"
+)
+
+// PGAvailable indicates whether a default PG database is available.
+func PGAvailable() bool {
+	db, err := sql.Open("postgres", dataSource)
+	if err != nil {
+		log.Printf("sql.Open(): %v", err)
+		return false
+	}
+	defer db.Close()
+	if err := db.Ping(); err != nil {
+		log.Printf("db.Ping(): %v", err)
+		return false
+	}
+	return true
+}
+
+// newEmptyDB creates a new, empty database.
+func newEmptyDB(ctx context.Context) (*sql.DB, error) {
+	db, err := sql.Open("postgres", dataSource)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a randomly-named database and then connect using the new name.
+	name := fmt.Sprintf("trl_%v", time.Now().UnixNano())
+
+	stmt := fmt.Sprintf("CREATE DATABASE %v", name)
+	if _, err := db.ExecContext(ctx, stmt); err != nil {
+		return nil, fmt.Errorf("error running statement %q: %v", stmt, err)
+	}
+
+	connStr := fmt.Sprintf("%s database=%s", dataSource, name)
+
+	db.Close()
+	db, err = sql.Open("postgres", connStr)
+	if err != nil {
+		return nil, err
+	}
+
+	return db, db.Ping()
+}
+
+// NewTrillianDB creates an empty database with the Trillian schema. The database name is randomly
+// generated.
+// NewTrillianDB is equivalent to Default().NewTrillianDB(ctx).
+func NewTrillianDB(ctx context.Context) (*sql.DB, error) {
+	db, err := newEmptyDB(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	sqlBytes, err := ioutil.ReadFile(trillianSQL)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, stmt := range strings.Split(sanitize(string(sqlBytes)), ";") {
```

Comment:
I guess it's not killer, but it looks like this would break if anyone ever put a semicolon that wasn't used as a line terminator anywhere in the file? 

`sanitize` attempts to fix this for comments (which was my first thought), so maybe it's good enough, just thought I'd mention it.